build-artifacts/project-template-gradle/src/main/java/com/tns/RuntimeHelper.java

```diff
@@ -53,157 +53,180 @@ public static Runtime initRuntime(Application app) {
+                        // see https://github.com/NativeScript/NativeScript/issues/4137 for reference
+                        boolean removePreviouslyInstalledAssets = true;
+                        aE.extractAssets(app, "app", outputDir, extractPolicy, removePreviouslyInstalledAssets);
+                        aE.extractAssets(app, "internal", outputDir, extractPolicy, removePreviouslyInstalledAssets);
+                        aE.extractAssets(app, "metadata", outputDir, extractPolicy, false);
 
-                // will extract snapshot of the device appropriate architecture
-                if (shouldExtractSnapshots) {
-                    if (logger.isEnabled()) {
-                        logger.write("Extracting snapshot blob");
-                    }
+                        boolean shouldExtractSnapshots = true;
 
-                    aE.extractAssets(app, "snapshots/" + Build.CPU_ABI, outputDir, extractPolicy, removePreviouslyInstalledAssets);
-                }
+                        // will extract snapshot of the device appropriate architecture
+                        if (shouldExtractSnapshots) {
+                            if (logger.isEnabled()) {
+                                logger.write("Extracting snapshot blob");
+                            }
 
-                extractPolicy.setAssetsThumb(app);
-            }
+                            aE.extractAssets(app, "snapshots/" + Build.CPU_ABI, outputDir, extractPolicy, removePreviouslyInstalledAssets);
+                        }
 
-            AppConfig appConfig = new AppConfig(appDir);
-
-            ClassLoader classLoader = app.getClassLoader();
-            File dexDir = new File(rootDir, "code_cache/secondary-dexes");
-            String dexThumb = null;
-            try {
-                dexThumb = Util.getDexThumb(app);
-            } catch (NameNotFoundException e) {
-                if (logger.isEnabled()) {
-                    logger.write("Error while getting current proxy thumb");
+                        extractPolicy.setAssetsThumb(app);
+                    } finally {
+                        extractionFrame.close();
+                    }
                 }
-                e.printStackTrace();
-            }
-
-            String nativeLibDir = null;
-            try {
-                nativeLibDir = app.getPackageManager().getApplicationInfo(appName, 0).nativeLibraryDir;
-            } catch (android.content.pm.PackageManager.NameNotFoundException e) {
-                e.printStackTrace();
-            }
 
-            boolean isDebuggable = Util.isDebuggableApp(app);
-            StaticConfiguration config = new StaticConfiguration(logger, appName, nativeLibDir, rootDir,
-                    appDir, classLoader, dexDir, dexThumb, appConfig, isDebuggable);
+                AppConfig appConfig = new AppConfig(appDir);
+                switch (appConfig.getProfilingMode()) {
```

Comment:
The following lines concerning instrumentation can be moved to the ManualInstrumentation class in a method you call from RuntimeHelper